packages/editor/src/components/dialogs/PublishModal.tsx

```diff
@@ -0,0 +1,271 @@
+  const { t } = useTranslation()
+
+  const locationMutation = useMutation(locationPath)
+
+  const submitLoading = useHookstate(false)
+  const errors = useHookstate(getDefaultErrors())
+
+  const name = useHookstate(location?.name || '')
+  const maxUsers = useHookstate(location?.maxUsersPerInstance || 20)
+
+  const scene = useHookstate((location ? location.sceneId : sceneID) ?? '')
+  const videoEnabled = useHookstate<boolean>(location?.locationSetting.videoEnabled || true)
+  const audioEnabled = useHookstate<boolean>(location?.locationSetting.audioEnabled || true)
+  const screenSharingEnabled = useHookstate<boolean>(location?.locationSetting.screenSharingEnabled || true)
+  const locationType = useHookstate(location?.locationSetting.locationType || 'public')
+
+  const scenes = useFind(staticResourcePath, {
+    query: {
+      paginate: false
+    }
+  })
+
+  const handleSubmit = async () => {
+    errors.set(getDefaultErrors())
+
+    if (!name.value) {
+      errors.name.set(t('admin:components.location.nameCantEmpty'))
+    }
+    if (!maxUsers.value) {
+      errors.maxUsers.set(t('admin:components.location.maxUserCantEmpty'))
+    }
+    if (!scene.value) {
+      errors.scene.set(t('admin:components.location.sceneCantEmpty'))
+    }
+    if (Object.values(errors.value).some((value) => value.length > 0)) {
+      return
+    }
+
+    submitLoading.set(true)
+
+    const locationData: LocationData = {
+      name: name.value,
+      slugifiedName: '',
+      sceneId: scene.value,
+      maxUsersPerInstance: maxUsers.value,
+      locationSetting: {
+        id: '',
+        locationId: '' as LocationID,
+        locationType: locationType.value,
+        audioEnabled: audioEnabled.value,
+        screenSharingEnabled: screenSharingEnabled.value,
+        faceStreamingEnabled: false,
+        videoEnabled: videoEnabled.value,
+        createdAt: '',
+        updatedAt: ''
+      },
+      isLobby: false,
+      isFeatured: false
+    }
+
+    try {
+      if (location?.id) {
+        await locationMutation.patch(location.id, locationData)
+      } else {
+        await locationMutation.create(locationData)
+      }
+      PopoverState.hidePopupover()
+    } catch (err) {
+      errors.serverError.set(err.message)
+    }
+    submitLoading.set(false)
+  }
+
+  const unpublishLocation = async () => {
+    if (location?.id) {
+      submitLoading.set(true)
+      try {
+        await locationMutation.remove(location.id)
+        PopoverState.hidePopupover()
+      } catch (err) {
+        errors.serverError.set(err.message)
+        submitLoading.set(false)
+      }
+    }
+  }
+
+  return (
+    <div className="relative z-50 max-h-[80vh] w-[50vw] bg-theme-surface-main">
+      <div className="relative rounded-lg shadow">
+        <ModalHeader
+          onClose={PopoverState.hidePopupover}
+          title={location?.id ? t('editor:toolbar.publishLocation.create') : t('editor:toolbar.publishLocation.update')}
```

Comment:
![image](https://github.com/EtherealEngine/etherealengine/assets/10975502/14270cc2-0929-4fc5-a576-a9f5d5ce1b1c)

After I unpusblished, its showing update text. And wrong `Publish Location` title is shown when a location is published.

Reply:
fixed!